### PR TITLE
PLG-183: Initialize the category edition page

### PR DESCRIPTION
**Pull request by olivierpontier**
**Description (for Contributor and Core Developer)**

**Definition Of Done (for Core Developer only)**

| Q                                 | A
| --------------------------------- | ---
| Added Specs                       | Todo
| Added legacy Behats               | Todo
| Added acceptance tests            | Todo
| Added integration tests           | Todo
| Changelog updated                 | Todo
| Review and 2 GTM                  | Todo
| Micro Demo to the PO (Story only) | Todo
| Migration script                  | -
| Tech Doc                          | -

`Todo`: Pending / Work in progress
`OK`: Done / Validated
`-`: Not needed

### CODE REVIEW

File: src/Akeneo/Pim/Structure/Bundle/Resources/workspaces/settings-ui/src/pages/categories/CategoryEditPage.tsx [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+  const userContext = useUserContext();
+  const {category, status, load} = useCategory(parseInt(categoryId));
+  const [categoryLabel, setCategoryLabel] = useState(`[${categoryId}]`);
+  const [treeLabel, setTreeLabel] = useState(translate('pim_enrich.entity.category.content.edit.default_tree_label'));
+
+  useSetPageTitle(translate('pim_title.pim_enrich_categorytree_edit', {'category.label': categoryLabel}));
+
+  const followSettingsIndex = () => router.redirect(router.generate('pim_enrich_attribute_index'));
+  const followCategoriesIndex = () => router.redirect(router.generate('pim_enrich_categorytree_index'));
+  const followCategoryTree = () => router.redirect(router.generate('pim_enrich_categorytree_tree', {id: 1}));
```
Comment (olivierpontier) [inline src/Akeneo/Pim/Structure/Bundle/Resources/workspaces/settings-ui/src/pages/categories/CategoryEditPage.tsx:25]:
```suggestion
    const followCategoryTree = () => {
    if (!category || category.root === undefined) {
      return;
    }
    router.redirect(router.generate('pim_enrich_categorytree_tree', {id: category.root}));
  }
```

File: src/Akeneo/Pim/Structure/Bundle/Resources/workspaces/settings-ui/src/pages/categories/CategoryEditPage.tsx [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+};
+
+const CategoryEditPage: FC = () => {
+  const {categoryId} = useParams<Params>();
+  const translate = useTranslate();
+  const router = useRouter();
+  const userContext = useUserContext();
+  const {category, status, load} = useCategory(parseInt(categoryId));
+  const [categoryLabel, setCategoryLabel] = useState(`[${categoryId}]`);
+  const [treeLabel, setTreeLabel] = useState(translate('pim_enrich.entity.category.content.edit.default_tree_label'));
```
Comment (olivierpontier) [inline src/Akeneo/Pim/Structure/Bundle/Resources/workspaces/settings-ui/src/pages/categories/CategoryEditPage.tsx:19]:
```suggestion
  const [treeLabel, setTreeLabel] = useState(translate('pim_enrich.entity.category.content.edit.default_tree_label'));
  const [tree, setTree] = useState<Category|null>(null);
```

File: src/Akeneo/Pim/Structure/Bundle/Resources/workspaces/settings-ui/src/pages/categories/CategoryEditPage.tsx [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+
+  useEffect(() => {
+    const tree = category && category.root ? category.root : category;
+
+    setCategoryLabel(category ? category.labels[userContext.get('uiLocale')] : `[${categoryId}]`);
+    setTreeLabel(
+      tree
+        ? tree.labels[userContext.get('uiLocale')]
+        : translate('pim_enrich.entity.category.content.edit.default_tree_label')
+    );
```
Comment (olivierpontier) [inline src/Akeneo/Pim/Structure/Bundle/Resources/workspaces/settings-ui/src/pages/categories/CategoryEditPage.tsx:39]:
```suggestion
        const rootCategory = category && category.root ? category.root : category;

    setCategoryLabel(category ? category.labels[userContext.get('uiLocale')] : `[${categoryId}]`);
    setTreeLabel(
      rootCategory
        ? rootCategory.labels[userContext.get('uiLocale')]
        : translate('pim_enrich.entity.category.content.edit.default_tree_label')
    );
    setTree(rootCategory)
```

**Review by LaurentPetard — APPROVED** [review]

File: src/Akeneo/Pim/Enrichment/Bundle/Controller/Ui/CategoryTreeController.php [hunk 4]
```diff
@@ -327,15 +332,18 @@ public function editAction(Request $request, $id)
...
+        if ($category->isRoot() === false) {
+            $rootCategory = $this->findCategory($category->getRoot());
+        }
+
+        $normalizedCategory = $this->normalizer->normalize($category, 'internal_api');
+        $normalizedCategory = array_merge($normalizedCategory, [
+            'root' => $rootCategory === null ? null : $this->normalizer->normalize($rootCategory, 'internal_api')
+        ]);
+
+        // @todo[PLG-94] remove unused Twig templates AkeneoPimEnrichmentBundle:CategoryTree:%s.html.twig
```
Comment (rouliane) [inline src/Akeneo/Pim/Enrichment/Bundle/Controller/Ui/CategoryTreeController.php:345]: Don't forget to handle this todo, I don't see the removal of this file in your PR

**Review by rouliane — APPROVED** [review]

File: src/Akeneo/Pim/Enrichment/Bundle/Controller/Ui/CategoryTreeController.php (hunk 4 above)
Comment (olivierpontier) [reply src/Akeneo/Pim/Enrichment/Bundle/Controller/Ui/CategoryTreeController.php:345]: It will be cleaned at the end of the development of the feature

**Pull request merged**